### ISSUE TITLE
Expose Flask  / UI  

### ISSUE THREAD

**Issue by h0rv4th** [ail-project/ail-framework#172]
Trying to expose the UI to the network I have modified core.cfg

```
#Proxying requests to the app
baseUrl = /
#Host to bind to
host = 0.0.0.0
#Flask server port
port = 7000
```
However, I cannot connect to the UI.

Understanding that there is a lack of documentation, this may not be the right way. 
That being the case, what is the correct way to expose the AIL UI?

**Comment by Terrtia** [comment]
Hi @h0rv4th !
you are editing the correct section.
To apply the configuration, you need to restart the Flask server

**Comment by fukusuket** [comment]
I was able to access the UI with the above method, thanks for the issue and comments :)

**Comment by h0rv4th** [comment]
Hello @Terrtia ,

I have configured Flask host to 0.0.0.0
then killed all with /ail-framework/bin# ./LAUNCH.sh -m / Option 7
And started all over again with ./LAUNCH.sh -L

Unfortunately it is not t work ... 

```
------------------------------------------------------------------

****************  AIL Successfully Updated  *****************

        * Launching Redis servers
        * Launching KVROCKS servers
There are several suitable screens on:
        129148.KVROCKS_AIL      (07/05/2023 08:57:26 AM)        (Detached)
        128900.KVROCKS_AIL      (07/05/2023 08:57:24 AM)        (Detached)
Use -S to specify a session.
        * Launching logging process
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
        * Checking configuration
Config File: Nothing to update
Config File: Nothing to update
        * Configuration up-to-date
        * Launching core scripts ...
/var/tools/ail-framework/bin /var/tools/ail-framework
dot -Tsvg /var/tools/ail-framework/doc/ail_queues.dot -o /var/tools/ail-framework/doc/ail_queues.svg
AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
/var/tools/ail-framework
        * Launching scripts
        * Launching Flask server               


```root@btrsvr-nessie-ail:/var/tools/ail-framework/bin# netstat -nltp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      6661/systemd-resolv
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      6646/sshd: /usr/sbi
tcp        0      0 127.0.0.1:6010          0.0.0.0:*               LISTEN      127093/sshd: 
tcp        0      0 127.0.0.1:6379          0.0.0.0:*               LISTEN      129134/redis-server
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6383          0.0.0.0:*               LISTEN      128912/./kvrocks/bu
tcp        0      0 127.0.0.1:6380          0.0.0.0:*               LISTEN      129141/redis-server
tcp        0      0 127.0.0.1:6381          0.0.0.0:*               LISTEN      129149/redis-server
tcp6       0      0 ::1:6010                :::*                    LISTEN      127093/sshd: 
tcp6       0      0 :::22                   :::*                    LISTEN      6646/sshd: /usr/sbi



Any clues?

**Comment by h0rv4th** [comment]
Any suggestions on how to identify the failure?

**Comment by h0rv4th** [comment]
@Terrtia 

logs/ail.log it is empty

```
logs/modules.log
2023-06-15 05:18:29,640 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-06-15 05:26:12,852 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-06-15 05:27:59,752 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-06-29 04:19:07,434 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-06-29 04:21:58,386 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-06-29 04:24:57,693 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
2023-07-05 08:57:29,385 root WARNING:AIL d871ec08-d526-44dc-bf31-02ff6d91f893 started
```

logs/flask.log does not exist



```
screen -r Flask_AIL

blueprints              Flask_server.py  static     update_thirdparty.sh
create_default_user.py  modules          templates
Traceback (most recent call last):
  File "/var/tools/ail-framework/var/www/./Flask_server.py", line 21, in <module>
    from pytaxonomies import Taxonomies
ModuleNotFoundError: No module named 'pytaxonomies'
```

**Comment by Terrtia** [comment]
Can you git pull and try to launch the flask server manually ?

```bash
cd ail-framework/
. ./AILENV/bin/activate
cd var/www
./Flask_server.py
```

**Comment by h0rv4th** [comment]
```
/var/tools/ail-framework/var/www# ./Flask_server.py
Traceback (most recent call last):
  File "/var/tools/ail-framework/var/www/./Flask_server.py", line 21, in <module>
    from pytaxonomies import Taxonomies
ModuleNotFoundError: No module named 'pytaxonomies'

```

**Comment by fukusuket** [comment]
@h0rv4th 
Excuse me for cutting in, I have also encountered the same error. (`No module named pytaxonomies`) 
At that time, it was because my SSH session had dropped prematurely(maybe?) and the installation hadn't completed.

If the installation was successful, the following `pip` command will be executed to install `PyTaxonomies`. However, in your environment `PyTaxonomyes` doesn't seem to be installed in your `virtualenv`.
https://github.com/ail-project/ail-framework/blob/2691000d0cdf15b90030a12c101af02bf22c9009/var/www/update_thirdparty.sh#L129
In my case, reinstalling fixed this error. Therefore, I recommend that you try reinstalling :)

**Comment by eortiz-nexplore** [comment]
@Terrtia 
blueprints              Flask_server.py  static     update_thirdparty.sh
create_default_user.py  modules          templates
VT submission is disabled
Module CreditCards initialized
Module: Onion Launched
Module Tools initialized
/var/tools/ail-framework/var/www/./Flask_server.py:82: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
  ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
Traceback (most recent call last):
  File "/var/tools/ail-framework/var/www/./Flask_server.py", line 83, in <module>
    ssl_context.load_cert_chain(certfile=os.path.join(Flask_dir, 'server.crt'), keyfile=os.path.join(Flask_dir, 'server.key'))
FileNotFoundError: [Errno 2] No such file or directory

**Comment by eortiz-nexplore** [comment]
@Terrtia 

`blueprints              Flask_server.py  static     update_thirdparty.sh
create_default_user.py  modules          templates
VT submission is disabled
Module CreditCards initialized
Module: Onion Launched
Module Tools initialized
/var/tools/ail-framework/var/www/./Flask_server.py:82: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
  ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
Traceback (most recent call last):
  File "/var/tools/ail-framework/var/www/./Flask_server.py", line 83, in <module>
    ssl_context.load_cert_chain(certfile=os.path.join(Flask_dir, 'server.crt'), keyfile=os.path.join(Flask_dir, 'server.key'))
FileNotFoundError: [Errno 2] No such file or directory

`

**Comment by eortiz-nexplore** [comment]
@Terrtia 

```

blueprints              Flask_server.py  static     update_thirdparty.sh
create_default_user.py  modules          templates
VT submission is disabled
Module CreditCards initialized
Module: Onion Launched
Module Tools initialized
/var/tools/ail-framework/var/www/./Flask_server.py:82: DeprecationWarning: ssl.PROTOCOL_TLSv1_2 is deprecated
  ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
Traceback (most recent call last):
  File "/var/tools/ail-framework/var/www/./Flask_server.py", line 83, in <module>
    ssl_context.load_cert_chain(certfile=os.path.join(Flask_dir, 'server.crt'), keyfile=os.path.join(Flask_dir, 'server.key'))
FileNotFoundError: [Errno 2] No such file or directory

```

**Comment by Terrtia** [comment]
Hi @eortiz-nexplore !

Your installation is incomplete. https://github.com/ail-project/ail-framework/blob/master/installing_deps.sh#L122
You need to relaunch the installer.

**Comment by KhaledEid020** [comment]
![image](https://github.com/ail-project/ail-framework/assets/46486159/5614ac7f-e80f-4dfc-a412-45446ce7327f)

Same problem

**Comment by Terrtia** [comment]
Hi @KhaledEid020 
It's not the same issue. 
You are using an older version of AIL. The issue with flask-login was fixed with fa6c50c164708ebfc378cb87c6fc2101a9ac4cf1

Please use the last version of the framework.

**Comment by KhaledEid020** [comment]
Can I have the OVA file to run without installation.
